RNA-seq/01-qc_trim_quant.md

```diff
@@ -0,0 +1,244 @@
+fastp does have [the functionality](https://github.com/OpenGene/fastp#per-read-cutting-by-quality-score) to perform trimming using a sliding window.
+
+_Note that there are two kinds of encoding for phred scores, phred33 and phred64. 
+FastQC guessed that the file for SRR585570 uses Sanger/Illumina 1.9 encoding (phred33). 
+If we had phred64 data, we'd use the `--phred64` flag._
+
+#### `--length_required`
+
+Trimming reads may result in short reads, which may effect gene expression estimates ([Williams et al. _BMC Bioinformatics._ 2016.](https://doi.org/10.1186/s12859-016-0956-2)).
+Using `--length_required 20` means that reads shorter than 20bp will be discarded (similar to what was used in Srivastava et al. above).
+
+#### `--report_title`
+
+When we look at the HTML report, it's helpful to quickly identify what sample the report is for. Using `--report title "SRR585570"` means that the report will be titled "SRR585570" rather than the default ("fastp report").
+
+#### `--json` and `--html`
+
+With these options, we're specifying where the [JSON](https://en.wikipedia.org/wiki/JSON) and HTML reports will be saved (in the `QC/gastric_cancer/fastp_output/` we created) and what the filenames will be. 
+Including the sample name in the filenames again may help us with project organization.
+
+**If we look at** `QC/gastric_cancer/fastp_output/SRR585570_fastp.json` **or the top of the HTML report, we can see that fastp reports certain metrics before and after filtering, which can be very useful in making analysis decisions.**
+
+### Quantification with Salmon
+
+We'll use [Salmon](https://combine-lab.github.io/salmon/) for quantifying transcript expression ([documentation](http://salmon.readthedocs.io/en/latest/)).
+Salmon ([Patro, et al. _Nature Methods._ 2017.](https://doi.org/10.1038/nmeth.4197)) is fast and requires very little memory, which makes it a great choice for running on your laptop during training.
+We can use the output for downstream analyses like differential expression analysis and clustering.
+We use Salmon in mapping mode, with mapping validation enabled, using the following command:
+
+```bash
+# We perform quantification on the files that have been trimmed
+# and use the index generated with -k 23, as this may "improve sensitivity"
+# per the Salmon documentation
+salmon quant -i index/Homo_sapiens/short_index \
+	-l A \
+	-1 data/fastq/gastric_cancer/SRR585570/SRR585570_fastp_1.fastq.gz \
+	-2 data/fastq/gastric_cancer/SRR585570/SRR585570_fastp_2.fastq.gz \
+	-o data/quant/gastric_cancer/SRR585570 \
+	--validateMappings --rangeFactorization 4 \
+	--gcBias --seqBias
+```
+
+Below, we'll walk through the arguments/options we used to run `salmon quant`.
+
+#### Transcriptome index: `-i`
```

Comment:
We could make a point to show them how arguments that are called the same thing `-i` can mean very different things in different programs or different functions and this emphasizes why it's important to reference the documentation.

Reply:
^ this may be a verbal point as opposed to written.

Reply:
Yes, I'm not going to make a written point about this.